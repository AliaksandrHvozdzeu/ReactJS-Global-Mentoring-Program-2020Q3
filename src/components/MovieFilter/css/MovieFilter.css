.filter-list {
    --white: #ffffff;
    color: var(--white);
    font-size: 1.2rem;
    letter-spacing: 0.1rem;
    margin: 0 0;
    padding: 0 0;
    box-sizing: border-box;
    display: flex;
    align-items: stretch;
    position: relative;
    height: 65px;
}

.filter-list * {
    letter-spacing: 0.1rem;
}

.filter-list::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    height: 3px;
    background: linear-gradient(180deg, rgba(122, 121, 121, 1) 0%, rgba(85, 85, 85, 1) 20%, rgba(0, 0, 0, 1) 100%);
    border-radius: 5px;
}

.filter-item {
    display: inline-block;
    width: 49%;
    margin: 0 0;
    padding: 0 0;
    box-sizing: border-box;
}

/* filter by genre */
.filter-item:not(:last-child) {
    margin-right: 2%;
}

.filter-item:first-child ul {
    display: flex;
    flex-direction: row;
    justify-content: start;
    flex-wrap: wrap;
}

.genre-filter-item {
    text-transform: uppercase;
}

.genre-filter-item:not(:last-child) {
    margin-right: 2rem;
}

.genre-filter-item, .genre-filter-item a {
    list-style-type: none;
    text-decoration: none;
    color: var(--white);
    position: relative;
}

.genre-filter-item a:hover::after {
    position: absolute;
    content: '';
    border-bottom: 3px solid var(--red);
    left: 0;
    right: 0;
    bottom: -19px;
    z-index: 2;
    border-radius: 1px;
}

.genre-filter-item {
    text-transform: uppercase;
}

/* filter by date */
.filter {
    text-align: right;
    text-transform: uppercase;
}

.filter-title {
    color: var(--white);
    font-weight: 100;
}

.filter-select {
    background-color: var(--grey-darker);
    color: var(--white);
    font-size: 1.2rem;
    outline: none;
    margin-left: 2rem;
}

.filter-select:hover {
    cursor: pointer;
}

.filter-select::after {
    content: '';
    border-top: 5px solid var(--red);
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-bottom: 5px solid transparent;
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: 2;
}
