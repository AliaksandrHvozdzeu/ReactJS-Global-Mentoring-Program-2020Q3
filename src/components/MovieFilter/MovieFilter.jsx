import React from 'react';
import PropTypes from 'prop-types';
import Constants from '../../constants';
import GenreFilter from '../MovieGenreFilter';
import MovieSorting from '../MovieSorting';
import './style.css';

const MovieFilter = ({ onFilterByGenre, onSorting }) => (
  <>
    <section className="filter-list">
      <div className="filter-item filter-by-genre">
        <GenreFilter genres={Constants.FILTERS} action={onFilterByGenre} />
      </div>
      <div className="filter-item filter">
        <MovieSorting title="SORT BY" options={Constants.SORTING} onAction={onSorting} />
      </div>
    </section>
  </>
);

MovieFilter.propTypes = {
  onFilterByGenre: PropTypes.func.isRequired,
  onSorting: PropTypes.func.isRequired
};

export default MovieFilter;
