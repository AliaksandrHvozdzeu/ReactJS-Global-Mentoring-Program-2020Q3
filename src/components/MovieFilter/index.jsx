import React from 'react';
import Constants from '../../constants';
import './style.css';
import GenreFilter from '../MovieGenreFilter';
import MovieSorting from '../MovieSorting';
import PropTypes from 'prop-types';
import MovieEdit from '../MovieEdit';

const MovieFilter = ({ onFilterByGenre, onSorting }) => (
  <>
    <section className="filter-list">
      <div className="filter-item filter-by-genre">
        <GenreFilter genres={Constants.FILTERS} action={onFilterByGenre} />
      </div>
      <div className="filter-item filter">
        <MovieSorting title="SORT BY" options={Constants.SORTING} onAction={onSorting} />
      </div>
    </section>
  </>
);

MovieEdit.propTypes = {
  onFilterByGenre: PropTypes.func.isRequired,
  onSorting: PropTypes.func.isRequired
};

export default MovieFilter;
