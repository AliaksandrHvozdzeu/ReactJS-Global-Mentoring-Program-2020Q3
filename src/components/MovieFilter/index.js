import MovieFilter from './js/MovieFilter';

export default MovieFilter;
