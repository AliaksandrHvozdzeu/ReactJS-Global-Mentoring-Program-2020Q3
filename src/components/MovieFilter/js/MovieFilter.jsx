import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import GenreFilter from '../../MovieGenreFilter';
import MovieSorting from '../../MovieSorting';
import filterActions from '../../../store/actions/filter';
import Constants from '../../../constants';
import '../css/MovieFilter.css';

const MovieFilter = ({ genres, onFilterByGenre, onSorting }) => {

  const onGenreFiltering = useCallback(
    (genre) => {
      if (genre === genres[0]) {
        onFilterByGenre(null);
      } else {
        onFilterByGenre(genre);
      }
    },
    [genres, onFilterByGenre],
  );

  return (
    <>
      <section className="filter-list">
        <div className="filter-item filter-by-genre">
          <GenreFilter genres={genres}
                       action={onGenreFiltering}/>
        </div>
        <div className="filter-item filter">
          <MovieSorting title="SORT BY"
                        options={Constants.SORTING}
                        onAction={onSorting}/>
        </div>
      </section>
    </>
  );

}

MovieFilter.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  onFilterByGenre: PropTypes.func.isRequired,
  onSorting: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  genres: state.movies.genres,
});

const mapDispatchToProps = (dispatch) => ({
  onFilterByGenre: (genre) => dispatch(filterActions.filterByGenre(genre)),
  onSorting: (order) => dispatch(filterActions.changeOrder(order)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MovieFilter);
