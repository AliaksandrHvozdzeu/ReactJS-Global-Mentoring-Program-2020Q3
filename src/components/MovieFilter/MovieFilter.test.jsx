import React from "react";
import rerender from "react-test-renderer";
import MovieFilter from "./js/MovieFilter";
import { jest } from "@jest/globals";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("when MovieFilter", () => {

  const initialStore = {
    movies: {
      "genres": [
        "Drama",
        "Romance"
      ]
    }
  };
  const store = configureStore([])(initialStore);
  store.dispatch = jest.fn();

  test("then snapshot created", () => {
    const genres = ["Comedy", "Crime", "Documentary", "Horror"];
    const onFilterByGenre = jest.fn();
    const onSorting = jest.fn();
    const component = rerender.create(
      <Provider store={store}>
        <MovieFilter genres={genres}
                     onFilterByGenre={onFilterByGenre}
                     onSorting={onSorting} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("then snapshot created when genres is empty", () => {
    const onFilterByGenre = jest.fn();
    const onSorting = jest.fn();
    const component = rerender.create(
      <Provider store={store}>
        <MovieFilter genres={[]}
                     onFilterByGenre={onFilterByGenre}
                     onSorting={onSorting} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
