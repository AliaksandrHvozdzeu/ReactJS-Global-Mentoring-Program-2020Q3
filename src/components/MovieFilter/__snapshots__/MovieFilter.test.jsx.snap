// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieFilter then snapshot created 1`] = `
<section
  className="filter-list"
>
  <div
    className="filter-item filter-by-genre"
  >
    <ul
      className="genre-filter"
    >
      <li
        className="genre-filter-item"
      >
        <button
          className="active"
          onClick={[Function]}
          type="button"
        >
          Drama
        </button>
      </li>
      <li
        className="genre-filter-item"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Romance
        </button>
      </li>
    </ul>
  </div>
  <div
    className="filter-item filter"
  >
    <label
      className="sorting-title"
      htmlFor="date-filter"
    >
      SORT BY
      <select
        className="sorting-select"
        id="date-filter"
        onChange={[Function]}
        value="Release"
      >
        <option
          className="sorting-item"
          value="Release"
        >
          Release
        </option>
        <option
          className="sorting-item"
          value="Runtime"
        >
          Runtime
        </option>
        <option
          className="sorting-item"
          value="Budget"
        >
          Budget
        </option>
        <option
          className="sorting-item"
          value="Revenue"
        >
          Revenue
        </option>
        <option
          className="sorting-item"
          value="Vote average"
        >
          Vote average
        </option>
        <option
          className="sorting-item"
          value="Vote count"
        >
          Vote count
        </option>
      </select>
    </label>
  </div>
</section>
`;

exports[`when MovieFilter then snapshot created when genres is empty 1`] = `
<section
  className="filter-list"
>
  <div
    className="filter-item filter-by-genre"
  >
    <ul
      className="genre-filter"
    >
      <li
        className="genre-filter-item"
      >
        <button
          className="active"
          onClick={[Function]}
          type="button"
        >
          Drama
        </button>
      </li>
      <li
        className="genre-filter-item"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Romance
        </button>
      </li>
    </ul>
  </div>
  <div
    className="filter-item filter"
  >
    <label
      className="sorting-title"
      htmlFor="date-filter"
    >
      SORT BY
      <select
        className="sorting-select"
        id="date-filter"
        onChange={[Function]}
        value="Release"
      >
        <option
          className="sorting-item"
          value="Release"
        >
          Release
        </option>
        <option
          className="sorting-item"
          value="Runtime"
        >
          Runtime
        </option>
        <option
          className="sorting-item"
          value="Budget"
        >
          Budget
        </option>
        <option
          className="sorting-item"
          value="Revenue"
        >
          Revenue
        </option>
        <option
          className="sorting-item"
          value="Vote average"
        >
          Vote average
        </option>
        <option
          className="sorting-item"
          value="Vote count"
        >
          Vote count
        </option>
      </select>
    </label>
  </div>
</section>
`;
